R/shiny.R

```diff
@@ -115,13 +115,20 @@ event_data <- function(
   }
   
   event <- match.arg(event)
-  # check to see if this event-source-priority combo is registered
   eventID <- paste(event, source, sep = "-")
   eventIDRegistered <- eventID %in% session$userData$plotlyShinyEventIDs
   if (!eventIDRegistered) {
-    stop(
-      "The '", event, "' event has not been registered for a source ID ",
-      "of '", source, "'. Please add `event_register(p, '", event, "')` to your plot `p`."
+    # If event_data() is requested before any (relevant) plotly
+    # graphs have rendered, we won't know if the relevant
+    # event-source combo is legitimate. As a result, we throw a 
+    # warning here to try to be useful in cases where you really
+    # are requesting an input value that is never registered, but
+    # without preventing valid code from running.
+    warning(
+      "The '", event, "' event tied a source ID of '", source, "' ",
+      "has not yet been registered. ", 
+      "If you encounter issues acquiring data from this event-source combo",
+      "Try adding `event_register(p, '", event, "')` to plot `p`"
     )
```

Comment:
This is gonna lead to a lot of false positive warnings (like in the app below), but I see no other great way around it. 

```r
library(shiny)

ui <- fluidPage(
    plotlyOutput("p")
)

server <- function(input, output, session) {
    
    observe({
        print(event_data("plotly_click"))
    })
    
    output$p <- renderPlotly({
        plot_ly(mtcars, x = ~mpg, y = ~wt)
    })
    
}

shinyApp(ui, server)
```

Reply:
Why's that--because the observer executes first?

Reply:
If that's the issue, you could defer the check/warning using `session$onFlushed`. That'll ensure that anybody who was going to register those events in the same "tick" will have done so.

Reply:
Oh, yes, good idea!